src/sentry/utils/appleconnect/itunes_connect.py

```diff
@@ -315,20 +321,21 @@ def get_dsym_url(
         f"{bundle_version}/details"
     )
 
-    logger.debug(f" GET {details_url}")
+    logger.debug(f"GET {details_url}")
 
     details_response = session.get(details_url)
 
-    if details_response.status_code == HTTPStatus.OK:
-        try:
-            data = details_response.json()
-            dsym_url = safe.get_path(data, "data", "dsymurl")
-            return dsym_url  # type: ignore
-        except Exception:
-            logger.info(
-                f"Could not obtain dsms info for app id={app_id}, bundle_short={bundle_short_version}, "
-                f"bundle={bundle_version}, platform={platform}",
-                exc_info=True,
-            )
-            return None
-    return None
+    # A non-OK status code will probably mean an expired token/session
+    if details_response.status_code != HTTPStatus.OK:
+        raise FailedFetchingDsymError
```

Comment:
It would be really cool if this could actually raise a specific `ITunesSessionExpiredException` separate from a generic "request failed" exception.  Do you think that's possible?